pkg/resources/task.go

```diff
@@ -348,24 +349,11 @@ func CreateTask(d *schema.ResourceData, meta interface{}) error {
 		precedingTasks := make([]sdk.SchemaObjectIdentifier, 0)
 		for _, dep := range after {
 			precedingTaskId := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, dep)
-			rootTasks, err := sdk.GetRootTasks(client.Tasks, ctx, precedingTaskId)
+			resumeSuspended, err := client.Tasks.TemporarilySuspendRootTasks(ctx, precedingTaskId, taskId)
+			defer func() { returnedErr = errors.Join(returnedErr, resumeSuspended()) }()
```

Comment:
first defer or first check for err and return? (in update there is the same ordering but different one is in delete)